### ISSUE TITLE
NPE in ServerRpcHandler.handleRpc():223

### ISSUE THREAD

**Issue by mvysny** [vaadin/multiplatform-runtime#80]
My app is failing with

```
Caused by: java.lang.NullPointerException
	at com.vaadin.server.communication.ServerRpcHandler.handleRpc(ServerRpcHandler.java:223)
	at com.vaadin.mpr.core.AbstractLegacyEmbed.setRequest(AbstractLegacyEmbed.java:148)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.vaadin.flow.server.communication.rpc.PublishedServerEventHandlerRpcHandler.invokeMethod(PublishedServerEventHandlerRpcHandler.java:200)
	... 54 more
```

This looks similar to #75 but it's caused by something else: in this case the `mprUIContent.getUI()` returns null since the `mprUIContent` never got attached. It's a mystery why `mprUIContent` never got attached yet it received a RPC call.

Vaadin 8.11.3, 14.3.7, MPR 2.0.5.

**Comment by mvysny** [comment]
Posting a couple of tips from @TatuLund 

* Does the project have custom components / views that override `attach()` method? Verify that `super.attach()` is called first.
* If there is some heavy operation done in `attach()` and `super.attach()` called after that, it could lead to to timing issues.
* Also have you verified that there has not been client side exception, that has lead to incomplete attach.

**Comment by mvysny** [comment]
I've discovered the cause of the problem: my UI forgot to call `super.init(vaadinRequest);` from its `init()` method.

We should harden MPR to fail with an informative error message in such cases: by default UI.init() does nothing, therefore Vaadin 8 apps can get away with not calling `super.init()`.

**Comment by caalador** [comment]
The documentation states that after migration you should not have a custom UI class left.

Then also noted in the advanced section: [Using a custom legacy UI class](https://vaadin.com/docs/v14/mpr/configuration/custom-ui.html)
There us a big note `You need to call super.init(request) if you need to override the init method`

**Comment by mvysny** [comment]
> The documentation states that after migration you should not have a custom UI class left.

The migration with MPR may be ongoing for entire year, which will pretty much require the custom UI class to be present in the project for the entire year.

> There us a big note `You need to call super.init(request) if you need to override the init method`

Yes. In a perfect world that remark would suffice. Unfortunately, people don't read manuals. Also, when migrating MPR, there are gazillion steps to take, and it's easy to forget to check everything. Also, the code base under migration is usually unfamiliar to me: my job is only to take the sources and migrate them. This makes it especially easy to miss things like this one.

It is therefore the job of the library to provide meaningful error messages and guide me what to do, in case I shoot myself in the foot unknowingly. Or are we happy with the same level of error reporting that DB2 and Oracle database does?

**Comment by caalador** [comment]
Issue is already fixed in #491 which had the wrong fixes reference
